### PR TITLE
Supermaster support, multiple slaves on the same mysql server 

### PR DESCRIPTION
**Pull request by tafkam**
I needed those changes to get docker-pdns running on Kubernetes with a supermaster and 2 slaves. All three PDNS instances can run on the same mysql backend. 
For working supermaster AXFR I had to add the proper supermaster hostname (=NS record entry) in the mysql supermasters table. 

If you are interested I can provide kubernetes example manifests for the docs (daemonset for supermaster, daemonset for slaves, deployment for pdns-admin, and a statefulset for mariadb serving all those components. DNS is reachable from external networks, and the admin-dashboard from ingress/auth proxy)

Added Env Vars:
- NODE_NAME: should be filled with pod env fieldRef.fieldPath: spec.nodeName. Used as database name suffix (e.g. slave_node3 with PDNS_gmysql_dbname="slave_")
- SUPERMASTER_HOSTS: list of hostnames to be written into the supermasters table. Each entry is matched by position with the values in SUPERMASTER_IPS

### CODE REVIEW

File: pdns/docker-entrypoint.sh [hunk 1]
```diff
@@ -12,17 +12,20 @@ fi
 : "${PDNS_gmysql_password:=${MYSQL_ENV_MYSQL_PASSWORD:-powerdns}}"
 : "${PDNS_gmysql_dbname:=${MYSQL_ENV_MYSQL_DATABASE:-powerdns}}"
 
-export PDNS_gmysql_host PDNS_gmysql_port PDNS_gmysql_user PDNS_gmysql_password PDNS_gmysql_dbname
+# use first part of node name as database name suffix
+if [ "${NODE_NAME:-}" ]; then
+    NODE_NAME=$(echo ${NODE_NAME} | sed -e 's/\..*//' -e 's/-//')
+fi
+PDNS_gmysql_dbname="${PDNS_gmysql_dbname}${NODE_NAME}"
```
Comment (pschiffe) [inline pdns/docker-entrypoint.sh]: This won't work if `NODE_NAME` is not defined. I think this line should be moved to the condition above.

File: pdns/docker-entrypoint.sh [hunk 2]
```diff
@@ -33,14 +36,30 @@ if [ "$MYSQL_NUM_TABLE" -eq 0 ]; then
...
+if [ "${PDNS_superslave:-no}" == "yes" ]; then
+    # Configure supermasters if needed
+    if [ "${SUPERMASTER_IPS:-}" ]; then
+        $MYSQL_COMMAND -D "$PDNS_gmysql_dbname" -e "TRUNCATE supermasters;"
+        MYSQL_INSERT_SUPERMASTERS=''
+        if [ "${SUPERMASTER_COUNT:-0}" == "0" ]; then
+            SUPERMASTER_COUNT=10
+        fi
+        i=1; while [ $i -le ${SUPERMASTER_COUNT} ]; do
+            SUPERMASTER_HOST=$(echo ${SUPERMASTER_HOSTS} | cut -d' ' -f${i})
```
Comment (pschiffe) [inline pdns/docker-entrypoint.sh]: This is not backward compatible, it needs to be `SUPERMASTER_HOST=$(echo ${SUPERMASTER_HOSTS:-} | cut -d' ' -f${i})`

**Comment by pschiffe** [comment]
Besides those two things I think it's good, thank you.

If you are able to contribute the kube examples, that would be awesome. I have it on my todo, but didn't have enough time to put it together.

**Comment by tafkam** [comment]
fixed those 2 issues. i'm gonna create a new pull requests for the manifests

**Comment by tafkam** [comment]
added manifests examples and documentation

**Comment by pschiffe** [comment]
Awesome, thank you!

**Pull request merged**